src/Nethermind/Nethermind.Xdc/Types/ExpCountDown.cs

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Threading;
+
+namespace Nethermind.Xdc.Types;
+
+public class ExpCountDown : IDisposable, IExpCountDown
+{
+    private int _initialDuration;
+    private int _base;
+    private int _maxExponent;
+    private int _currentExponent;
+    private Timer? _timer;
+    private Action? _callback;
+
+    public ExpCountDown(int initialDuration, int @base, int maxExponent)
+        => SetParams(initialDuration, @base, maxExponent, false);
+
+    public void SetParams(int initialDuration, int @base, int maxExponent, bool shouldScheduleNext = true)
+    {
+        if (@base <= 1) throw new ArgumentException("Base must be > 1", nameof(@base));
+        if (maxExponent < 0) throw new ArgumentException("MaxExponent must be >= 0", nameof(maxExponent));
+
+        _initialDuration = initialDuration;
+        _base = @base;
+        _maxExponent = maxExponent;
+        _currentExponent = 0;
+
+        if (_callback != null && shouldScheduleNext)
```

Comment:
use `is null`/`is not null`
```suggestion
        if (_callback is not null && shouldScheduleNext)
```